R/RLearner_surv_coxph.R

```diff
@@ -26,14 +26,23 @@ makeRLearner.surv.coxph = function() {
 trainLearner.surv.coxph = function(.learner, .task, .subset, .weights = NULL,  ...) {
   f = getTaskFormula(.task)
   data = getTaskData(.task, subset = .subset)
-  if (is.null(.weights)) {
-    survival::coxph(formula = f, data = data, ...)
-  } else  {
+  if (.learner$predict.type == "response") {
     survival::coxph(formula = f, data = data, weights = .weights, ...)
+  } else  {
+    model = survival::coxph(formula = f, data = data, weights = .weights, ...)
+    model = attachTrainingTime(model, task, data)
+    model
   }
 }
 
 #' @export
 predictLearner.surv.coxph = function(.learner, .model, .newdata, ...) {
-  predict(.model$learner.model, newdata = .newdata, type = "lp", ...)
+  if (.learner$predict.type == "response") {
+    predict(.model$learner.model, newdata = .newdata, type = "lp", ...)
+  } else {
+    preds = predict(.model$learner.model, newdata = .newdata, type = "lp", ...)
+    train.times = c(0, sort(unique(.model$learner.model$times)))
```

Comment:
This allows to have a vector with the first two elements being `0`. Is this intended?